packages/firestore/test/unit/specs/limbo_spec.test.ts

```diff
@@ -1171,4 +1171,278 @@ describeSpec('Limbo Documents:', [], () => {
+
+      const docA = doc('collection/a', 1000, { key: 'a', included: false });
+      const docA2 = doc('collection/a', 1007, { key: 'a', included: true });
+      const docC = doc('collection/c', 1002, { key: 'c', included: true });
+
+      const limboQueryC = newQueryForPath(docC.key.path);
+
+      return (
+        spec()
+          // onSnapshot(fullQuery) - fullQuery is listening to documents in the collection for the full test
+          .userListens(fullQuery)
+          .watchAcksFull(fullQuery, 1001, docA, docC)
+          .expectEvents(fullQuery, {
+            fromCache: false,
+            added: [docA, docC]
+          })
+
+          // docC was deleted, this puts docC in limbo and causes a snapshot from cache (metadata-only change)
+          .watchSends({ removed: [fullQuery] }, docC)
+          .watchCurrents(fullQuery, 'resume-token-1002')
+          .watchSnapshots(1002)
+          .expectLimboDocs(docC.key)
+          .expectEvents(fullQuery, {
+            fromCache: true
+          })
+
+          // User begins getDocs(filterQuery)
+          .userListensForGet(filterQuery)
+
+          // getDocs(filterQuery) will not resolve on the snapshot from cache
+          .expectEvents(filterQuery, {
+            fromCache: true,
+            added: [docC]
+          })
+
+          // Watch acks limbo and filter queries
+          .watchAcks(limboQueryC)
+          .watchAcks(filterQuery)
+
+          // Watch responds to limboQueryC - docC was deleted
+          .watchSends({ affects: [limboQueryC] })
+          .watchCurrents(limboQueryC, 'resume-token-1009')
+          .watchSnapshots(1009, [limboQueryC, fullQuery])
+
+          // However, docC is still in limbo because there has not been a global snapshot
+          .expectLimboDocs(docC.key)
+
+          // Rapid events of document update and delete caused by application
+          .watchSends({ removed: [filterQuery] }, docA)
+          .watchCurrents(filterQuery, 'resume-token-1004')
+          .watchSends({ affects: [filterQuery] }, docC)
+          .watchCurrents(filterQuery, 'resume-token-1005')
+          .watchSends({ removed: [filterQuery] }, docC)
+          .watchSends({ affects: [filterQuery] }, docA2)
+          .watchCurrents(filterQuery, 'resume-token-1007')
+
+          .watchSnapshots(1010, [fullQuery, limboQueryC])
+
+          // All changes are current and we get a global snapshot
+          .watchSnapshots(1010, [])
+
+          // Now docC is out of limbo
+          .expectLimboDocs()
+          .expectEvents(fullQuery, {
+            fromCache: false,
+            modified: [docA2],
+            removed: [docC]
+          })
+          // Now getDocs(filterQuery) can be resolved
+          .expectEvents(filterQuery, {
+            fromCache: false,
+            removed: [docC],
+            added: [docA2]
+          })
+      );
+    }
+  );
+
+  specTest('Fix #8474 - Watch sends deletedDoc on limbo resolution', [], () => {
```

Comment:
nit: please re-word this test case's name too... by reading it's name I'm not clear what it's testing or what is expected.

Reply:
I've cleaned up the test name and some of the step names to make the test more readable.